client/lib/load-script/callback-handler.js

```diff
@@ -0,0 +1,90 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+const debug = debugFactory( 'lib/load-script/callback-handler' );
+
+/**
+ * Module variables
+ */
+const callbacksForURLsInProgress = new Map();
+
+export function getCallbackMap() {
+	return callbacksForURLsInProgress;
+}
+
+export function isLoading( url ) {
+	return callbacksForURLsInProgress.has( url );
+}
+
+export function addScriptCallback( url, callback ) {
+	if ( isLoading( url ) ) {
+		debug( `Adding a callback for an existing script from "${ url }"` );
+		callbacksForURLsInProgress.get( url ).push( callback );
+	} else {
+		debug( `Adding a callback for a new script from "${ url }"` );
+		callbacksForURLsInProgress.set( url, [ callback ] );
+	}
+}
+
+export function removeScriptCallback( url, callback ) {
+	debug( `Removing a known callback for a script from "${ url }"` );
+
+	if ( ! callbacksForURLsInProgress.has( url ) ) {
+		return;
+	}
+
+	const index = callbacksForURLsInProgress.get( url ).indexOf( callback );
+
+	if ( -1 === index ) {
+		return;
+	}
```

Comment:
is `index` used somewhere? should we just check for membership and indicate that intention?

```js
if ( ! includes( getCallbacks().get( url ), callback ) ) {
	return;
}
```

Reply:
The `index` is used to splice out the callback from the array at `callbackMap.get( url )`, but I think checking membership explicitly will make the code more readable.